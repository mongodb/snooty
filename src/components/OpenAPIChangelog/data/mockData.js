export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
