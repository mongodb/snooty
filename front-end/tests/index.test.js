import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Index from '../src/templates/index';

// data for this component
import mockData from '../tests/data/index.test.json';

it('renders correctly', () => {
  const tree = renderer
    .create(<Index pageContext={ { __refDocMapping: mockData } } />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
