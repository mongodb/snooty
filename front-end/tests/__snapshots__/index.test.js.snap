// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="content"
>
  <div
    className="guide-category-list"
  >
    <div
      className="section"
      id="guides"
    >
      <div>
        <h1>
          Guides
          <a
            className="headerlink"
            href="#guides"
            title="Permalink to this headline"
          >
            Â¶
          </a>
        </h1>
        <section
          className="guide-category"
        >
          <div
            className="guide-category__title guide-category__title--getting-started"
          >
            Getting Started
          </div>
          <div
            className="guide-category__guides"
          >
            <a
              className="guide guide--regular"
              href="server/introduction"
            >
              <section
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="guide__title"
                >
                  Structure your Data for MongoDB
                </div>
                <ul
                  className="guide__body"
                >
                  
                </ul>
                <ul
                  className="guide__pills"
                />
                <div
                  className="guide__time"
                >
                  5min
                </div>
              </section>
            </a>
            <a
              className="guide guide--regular"
              href="server/install"
            >
              <section
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="guide__title"
                >
                  Install MongoDB
                </div>
                <ul
                  className="guide__body"
                >
                  
                </ul>
                <ul
                  className="guide__pills"
                />
                <div
                  className="guide__time"
                >
                  5min
                </div>
              </section>
            </a>
            <div
              className="guide guide--jumbo guide--expanded"
            >
              <section
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="guide__title"
                >
                  MongoDB in the Cloud
                </div>
                <ul
                  className="guide__body"
                >
                  <li
                    className="guide__entry"
                  >
                    <a
                      href="cloud/atlas"
                    >
                      Create an Atlas Account and Cluster
                    </a>
                  </li>
                  <li
                    className="guide__entry"
                  >
                    <a
                      href="cloud/connectionstring"
                    >
                      Set Up Atlas Connectivity
                    </a>
                  </li>
                </ul>
                <ul
                  className="guide__pills"
                />
                <div
                  className="guide__time"
                >
                  
                </div>
              </section>
            </div>
            <div
              className="guide guide--jumbo guide--expanded"
            >
              <section
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="guide__title"
                >
                  Migrate to MongoDB Atlas
                </div>
                <ul
                  className="guide__body"
                >
                  <li
                    className="guide__entry"
                  >
                    <a
                      href="cloud/migrate-from-aws-to-atlas"
                    >
                      Migrate a MongoDB Replica Set from AWS to MongoDB Atlas
                    </a>
                  </li>
                  <li
                    className="guide__entry"
                  >
                    <a
                      href="cloud/migrate-from-compose"
                    >
                      Migrate from Compose to MongoDB Atlas
                    </a>
                  </li>
                  <li
                    className="guide__entry"
                  >
                    <a
                      href="cloud/migrate-from-mlab"
                    >
                      Migrate from mLab to MongoDB Atlas
                    </a>
                  </li>
                </ul>
                <ul
                  className="guide__pills"
                />
                <div
                  className="guide__time"
                >
                  
                </div>
              </section>
            </div>
            <a
              className="guide guide--regular"
              href="server/import"
            >
              <section
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="guide__title"
                >
                  Import Data into MongoDB
                </div>
                <ul
                  className="guide__body"
                >
                  
                </ul>
                <ul
                  className="guide__pills"
                />
                <div
                  className="guide__time"
                >
                  5min
                </div>
              </section>
            </a>
            <a
              className="guide guide--regular"
              href="server/drivers"
            >
              <section
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="guide__title"
                >
                  Connect to MongoDB
                </div>
                <ul
                  className="guide__body"
                >
                  
                </ul>
                <ul
                  className="guide__pills"
                />
                <div
                  className="guide__time"
                >
                  5min
                </div>
              </section>
            </a>
            <div
              className="guide guide--jumbo guide--expanded"
            >
              <section
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="guide__title"
                >
                  CRUD Guides: Create, Read, Update, and Delete Data
                </div>
                <ul
                  className="guide__body"
                >
                  <li
                    className="guide__entry"
                  >
                    <a
                      href="server/insert"
                    >
                      Insert Data into MongoDB
                    </a>
                  </li>
                  <li
                    className="guide__entry"
                  >
                    <a
                      href="server/read"
                    >
                      Read Data from MongoDB
                    </a>
                  </li>
                  <li
                    className="guide__entry"
                  >
                    <a
                      href="server/read_queries"
                    >
                      Read Data from MongoDB With Queries
                    </a>
                  </li>
                  <li
                    className="guide__entry"
                  >
                    <a
                      href="server/read_operators"
                    >
                      Read Data using Operators and Compound Queries
                    </a>
                  </li>
                  <li
                    className="guide__entry"
                  >
                    <a
                      href="server/update"
                    >
                      Update Data in MongoDB
                    </a>
                  </li>
                  <li
                    className="guide__entry"
                  >
                    <a
                      href="server/delete"
                    >
                      Delete Data from MongoDB
                    </a>
                  </li>
                </ul>
                <ul
                  className="guide__pills"
                />
                <div
                  className="guide__time"
                >
                  
                </div>
              </section>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
